### DISCUSSION TITLE
Operator 'in' for collections

### DISCUSSION THREAD

**Discussion by jos3s** [dotnet/csharplang#8146] [Language Ideas]
It would be interesting to have the “in” operator as syntactic sugar for the “Contains” operation of collections

```csharp
 if (collection.Contains(value)) {}
```
Instead of writing it as it is above, we would write it like this:
```csharp
if (value in collection) {}
```

**Comment by HaloFour** [comment]
See: #420 #1665

**Reply by jos3s** [reply]
As far as I can see, none of these discussions have advanced since 2018/2019 or even analyzed the 'in' operator as 'Contains' as a possible initial version of their respective implementations

**Reply by chrisoverzero** [reply]
Exactly that was proposed in [the final post to #1665](https://github.com/dotnet/csharplang/discussions/1665#discussioncomment-129773). It didn't get traction.

**Reply by CyrusNajmabadi** [reply]
> As far as I can see, none of these discussions have advanced since 2018/2019 or even analyzed the 'in' operator as 'Contains' as a possible initial version of their respective implementations

Correct. No one has been interested in advancing this.

**Comment by huoyaoyuan** [comment]
Also #4255 #5708, and something I replied much earlier but I can't find.

**Comment by Joe4evr** [comment]
[As stated in the Framework Design Guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/operator-overloads):
> ❌ DO NOT be cute when defining operator overloads.

I take that as also including proposing a currently non-existant operator for an operation that's already trivially achievable.
